common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -1201,20 +1204,33 @@ public ICacheRecord getAccountWithAggregatedAccountDataByLocalAccountId(
 
         final List<BrokerApplicationMetadata> allMetadata = mApplicationMetadataCache.getAll();
 
+        // TODO - Everything inside this loop can be parallelized... should it be?
```

Comment:
We should NOT do this until we can measure it.  If there aren't enough execution units to manage it, it will be slower than not.

Reply:
ACK - holding....

Reply:
I believe we don't need to wait for Paul's perf work to measure this specific thing...that's more E2E perf measurement...over here I think we can just run a test with both old and new code and just capture numbers in the test and assert that the new code takes < time than old code (we can probably capture and assert on a percentage increase)

Also this is a sort of change that I think is also easy to flight...so if we have ability to do that now then we should try that too.

Reply:
We don't really have that ability, we could set it up, but nothing is populating the flight info right now, and we're not integrate with any metrics system that I know of.  The only thing I'm cautioning on is the parallelization.  Do we really know what we're getting into here, and will it actually have the effect we want?  If we want to tackle this, - don't create new thread pools to handle it each time, don't used a cached pool, we MUST measure it, and we need to monitor it.

Reply:
Reverted these changes in 6c0625b886331da1e58d55f8604965b22d81fc4d -- not going to try this until we can measure/evaluate